File: Assignment_38/Program4.cpp
// Write generic program to accept N values from user and return largest values.

#include<iostream>
using namespace std;

template <class T>

T Max(T *arr,int iSize)
{
    T max = arr[0];
    int i = 0;

    for(i = 0;i < iSize;i++)
    {
        if(arr[i] > max)
        {
            max = arr[i];
        }
    }
    return max;
}

int main()
{
    int arr[] = {10,20,30,40,50};
    float brr[] = {10.0f,3.7f,9.8f,8.7f};

    int iRet = Max(arr,5);
    cout<<"Integer largest value is: "<<iRet<<"\n";

    float fRet = Max(brr,4);
    cout<<"Float largest value is: "<<fRet<<"\n";

    return 0;
}

File: Assignment_53/Program4.java
/* Write a java program which accept two arrays from user and copy the contents of that array into another array & return new array.*/

import java.util.*;

class MyArray
{
    public int[] CopyArray(int arr[])
    {
        int i = 0;
        int newArr[] = new int[arr.length];

        for (i = 0;i < arr.length;i++)
        {
            newArr[i] = arr[i];
        }

        return newArr;
    }
}

class Program4
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int i = 0;

        System.out.println("Enter elements of first array:");
        String line1 = sobj.nextLine();
        String[] tokens1 = line1.trim().split("\\s+");
        int size1 = tokens1.length;
        int arr1[] = new int[size1];
        for (i = 0;i < size1;i++)
        {
            arr1[i] = Integer.parseInt(tokens1[i]);
        }

        System.out.println("Enter elements of second array:");
        String line2 = sobj.nextLine();
        String[] tokens2 = line2.trim().split("\\s+");
        int size2 = tokens2.length;
        int arr2[] = new int[size2];
        for (i = 0;i < size2;i++)
        {
            arr2[i] = Integer.parseInt(tokens2[i]);
        }

        MyArray mobj = new MyArray();
        
        int[] newArr1 = mobj.CopyArray(arr1);

        System.out.print("New first array: ");
        for (int num : newArr1)
        {
            System.out.print(num+"\t");
        }
        System.out.println();

        int[] newArr2 = mobj.CopyArray(arr2);

        System.out.print("New second array: ");
        for (int num : newArr2)
        {
            System.out.print(num+"\t");
        }
        System.out.println();

        sobj.close();
    }
}